[BUG] Octal escaping can consume too many digits and does not match

**Describe the bug**
When escaping octal digits for characters in regular expressions, the parser seems to consume a trailing digit that should not be part of the octal number being escaped.

**Steps/Code to reproduce bug**
Sample Python code:
```
>>> import cudf
>>> s = cudf.Series([",3Z\r\r"])
>>> s.str.contains("(\\0543)", regex=True)
0    False
dtype: bool
>>> s = cudf.Series([",Z\r\r"])
>>> s.str.contains("(\\054Z)", regex=True)
0    True
dtype: bool
```
Here `\054` is the octal representation of a comma `,`.  In the first case the comma is next to `3` but the regex is not matching, but in the second case the comma is next to `Z` and the regex is matching.

**Expected behavior**
Both `s.str.contains(...)` calls should return a Series with a single `True` value.

**Environment overview (please complete the following information)**
 - Environment location: Bare-metal
 - Method of cuDF install: from source

**Environment details**
```
<details><summary>Click here to see environment details</summary><pre>
     
     **git***
     commit 0581975e0132bd4189b579595e480a94d94a7917 (HEAD -> branch-22.04)
     Author: Nghia Truong <ttnghia@users.noreply.github.com>
     Date:   Thu Feb 3 15:12:56 2022 -0700
     
     Fix compile error in `binaryop/compiled/util.cpp` (#10209)
     
     This PR fixes a compile error in `binaryop/compiled/util.cpp`, when a function with return type `std::optional` return `{}` instead of `std::nullopt`. As such, nvcc 11.5/g++-10.3 yell:
     ```
     ../src/binaryop/compiled/util.cpp: In function 'std::optional<cudf::data_type> cudf::binops::compiled::get_common_type(cudf::data_type, cudf::data_type, cudf::data_type)':
     ../src/binaryop/compiled/util.cpp:48:15: error: '<anonymous>' may be used uninitialized in this function [-Werror=maybe-uninitialized]
     48 |       return {};
     |               ^
     ```
     
     Authors:
     - Nghia Truong (https://github.com/ttnghia)
     
     Approvers:
     - Robert Maynard (https://github.com/robertmaynard)
     - Karthikeyan (https://github.com/karthikeyann)
     - MithunR (https://github.com/mythrocks)
     - Bradley Dice (https://github.com/bdice)
     
     URL: https://github.com/rapidsai/cudf/pull/10209
     **git submodules***
     
     ***OS Information***
     DISTRIB_ID=Ubuntu
     DISTRIB_RELEASE=18.04
     DISTRIB_CODENAME=bionic
     DISTRIB_DESCRIPTION="Ubuntu 18.04.5 LTS"
     NAME="Ubuntu"
     VERSION="18.04.5 LTS (Bionic Beaver)"
     ID=ubuntu
     ID_LIKE=debian
     PRETTY_NAME="Ubuntu 18.04.5 LTS"
     VERSION_ID="18.04"
     HOME_URL="https://www.ubuntu.com/"
     SUPPORT_URL="https://help.ubuntu.com/"
     BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
     PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
     VERSION_CODENAME=bionic
     UBUNTU_CODENAME=bionic
     Linux c240m5-01 5.4.0-91-generic #102~18.04.1-Ubuntu SMP Thu Nov 11 14:46:36 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
     
     ***GPU Information***
     Fri Feb  4 12:01:39 2022
     +-----------------------------------------------------------------------------+
     | NVIDIA-SMI 495.29.05    Driver Version: 495.29.05    CUDA Version: 11.5     |
     |-------------------------------+----------------------+----------------------+
     | GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
     | Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
     |                               |                      |               MIG M. |
     |===============================+======================+======================|
     |   0  Tesla T4            On   | 00000000:19:00.0 Off |                    0 |
     | N/A   41C    P0    26W /  70W |    406MiB / 15109MiB |      0%      Default |
     |                               |                      |                  N/A |
     +-------------------------------+----------------------+----------------------+
     |   1  Tesla T4            On   | 00000000:5E:00.0 Off |                    0 |
     | N/A   33C    P8     9W /  70W |      7MiB / 15109MiB |      0%      Default |
     |                               |                      |                  N/A |
     +-------------------------------+----------------------+----------------------+
     |   2  Tesla T4            On   | 00000000:86:00.0 Off |                    0 |
     | N/A   32C    P8     9W /  70W |      7MiB / 15109MiB |      0%      Default |
     |                               |                      |                  N/A |
     +-------------------------------+----------------------+----------------------+
     |   3  Tesla T4            On   | 00000000:AF:00.0 Off |                    0 |
     | N/A   32C    P8     9W /  70W |      7MiB / 15109MiB |      0%      Default |
     |                               |                      |                  N/A |
     +-------------------------------+----------------------+----------------------+
     
     +-----------------------------------------------------------------------------+
     | Processes:                                                                  |
     |  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
     |        ID   ID                                                   Usage      |
     |=============================================================================|
     |    0   N/A  N/A      1942      G   /usr/lib/xorg/Xorg                  4MiB |
     |    0   N/A  N/A     50034      C   ...8-openjdk-amd64//bin/java      399MiB |
     |    1   N/A  N/A      1942      G   /usr/lib/xorg/Xorg                  4MiB |
     |    2   N/A  N/A      1942      G   /usr/lib/xorg/Xorg                  4MiB |
     |    3   N/A  N/A      1942      G   /usr/lib/xorg/Xorg                  4MiB |
     +-----------------------------------------------------------------------------+
     
     ***CPU***
     Architecture:        x86_64
     CPU op-mode(s):      32-bit, 64-bit
     Byte Order:          Little Endian
     CPU(s):              72
     On-line CPU(s) list: 0-71
     Thread(s) per core:  2
     Core(s) per socket:  18
     Socket(s):           2
     NUMA node(s):        2
     Vendor ID:           GenuineIntel
     CPU family:          6
     Model:               85
     Model name:          Intel(R) Xeon(R) Gold 6154 CPU @ 3.00GHz
     Stepping:            4
     CPU MHz:             1200.107
     CPU max MHz:         3700.0000
     CPU min MHz:         1200.0000
     BogoMIPS:            6000.00
     Virtualization:      VT-x
     L1d cache:           32K
     L1i cache:           32K
     L2 cache:            1024K
     L3 cache:            25344K
     NUMA node0 CPU(s):   0-17,36-53
     NUMA node1 CPU(s):   18-35,54-71
     Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cdp_l3 invpcid_single pti intel_ppin ssbd mba ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm cqm mpx rdt_a avx512f avx512dq rdseed adx smap clflushopt clwb intel_pt avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req pku ospke md_clear flush_l1d
     
     ***CMake***
     /home/navink/miniconda3/envs/cudf_dev/bin/cmake
     cmake version 3.21.3
     
     CMake suite maintained and supported by Kitware (kitware.com/cmake).
     
     ***g++***
     /usr/bin/g++
     g++ (Ubuntu 9.3.0-11ubuntu0~18.04.1) 9.3.0
     Copyright (C) 2019 Free Software Foundation, Inc.
     This is free software; see the source for copying conditions.  There is NO
     warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
     
     
     ***nvcc***
     /usr/local/cuda/bin/nvcc
     nvcc: NVIDIA (R) Cuda compiler driver
     Copyright (c) 2005-2021 NVIDIA Corporation
     Built on Thu_Nov_18_09:45:30_PST_2021
     Cuda compilation tools, release 11.5, V11.5.119
     Build cuda_11.5.r11.5/compiler.30672275_0
     
     ***Python***
     /home/navink/miniconda3/envs/cudf_dev/bin/python
     Python 3.8.12
     
     ***Environment Variables***
     PATH                            : /usr/local/cuda/bin:/home/navink/spark/spark-3.2.0-bin-hadoop3.2/bin:/usr/lib/jvm/java-8-openjdk-amd64/bin:/home/navink/.local/bin:/home/navink/miniconda3/envs/cudf_dev/bin:/home/navink/miniconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
     LD_LIBRARY_PATH                 :
     NUMBAPRO_NVVM                   :
     NUMBAPRO_LIBDEVICE              :
     CONDA_PREFIX                    : /home/navink/miniconda3/envs/cudf_dev
     PYTHON_PATH                     :
     
     ***conda packages***
     /home/navink/miniconda3/condabin/conda
     # packages in environment at /home/navink/miniconda3/envs/cudf_dev:
     #
     # Name                    Version                   Build  Channel
     _libgcc_mutex             0.1                 conda_forge    conda-forge
     _openmp_mutex             4.5                      1_llvm    conda-forge
     abseil-cpp                20210324.2           h9c3ff4c_0    conda-forge
     alabaster                 0.7.12                     py_0    conda-forge
     appdirs                   1.4.4              pyh9f0ad1d_0    conda-forge
     argon2-cffi               21.3.0             pyhd8ed1ab_0    conda-forge
     argon2-cffi-bindings      21.2.0           py38h497a2fe_1    conda-forge
     arrow-cpp                 5.0.0           py38h579a05f_22_cuda    conda-forge
     arrow-cpp-proc            3.0.0                      cuda    conda-forge
     asttokens                 2.0.5              pyhd8ed1ab_0    conda-forge
     attrs                     21.4.0             pyhd8ed1ab_0    conda-forge
     aws-c-cal                 0.5.11               h95a6274_0    conda-forge
     aws-c-common              0.6.2                h7f98852_0    conda-forge
     aws-c-event-stream        0.2.7               h3541f99_13    conda-forge
     aws-c-io                  0.10.5               hfb6a706_0    conda-forge
     aws-checksums             0.1.11               ha31a3da_7    conda-forge
     aws-sdk-cpp               1.8.186              hb4091e7_3    conda-forge
     babel                     2.9.1              pyh44b312d_0    conda-forge
     backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
     backports                 1.0                        py_2    conda-forge
     backports-zoneinfo        0.2.1                    pypi_0    pypi
     backports.functools_lru_cache 1.6.4              pyhd8ed1ab_0    conda-forge
     beautifulsoup4            4.10.0             pyha770c72_0    conda-forge
     binutils_impl_linux-64    2.36.1               h193b22a_2    conda-forge
     black                     19.10b0                    py_4    conda-forge
     bleach                    4.1.0              pyhd8ed1ab_0    conda-forge
     bokeh                     2.4.2            py38h578d9bd_0    conda-forge
     brotlipy                  0.7.0           py38h497a2fe_1003    conda-forge
     bzip2                     1.0.8                h7f98852_4    conda-forge
     c-ares                    1.18.1               h7f98852_0    conda-forge
     ca-certificates           2021.10.8            ha878542_0    conda-forge
     cachetools                5.0.0              pyhd8ed1ab_0    conda-forge
     certifi                   2021.10.8        py38h578d9bd_1    conda-forge
     cffi                      1.15.0           py38h3931269_0    conda-forge
     cfgv                      3.3.1              pyhd8ed1ab_0    conda-forge
     charset-normalizer        2.0.10             pyhd8ed1ab_0    conda-forge
     clang                     11.1.0               ha770c72_1    conda-forge
     clang-11                  11.1.0          default_ha53f305_1    conda-forge
     clang-tools               11.1.0          default_ha53f305_1    conda-forge
     clangxx                   11.1.0          default_ha53f305_1    conda-forge
     click                     8.0.3            py38h578d9bd_1    conda-forge
     cloudpickle               2.0.0              pyhd8ed1ab_0    conda-forge
     cmake                     3.21.3               h8897547_0    conda-forge
     cmake_setuptools          0.1.3                      py_0    rapidsai
     colorama                  0.4.4              pyh9f0ad1d_0    conda-forge
     commonmark                0.9.1                      py_0    conda-forge
     cryptography              36.0.1           py38h3e25421_0    conda-forge
     cuda-python               11.6.0           py38h3fd9d12_0    nvidia
     cudatoolkit               11.5.0               h36ae40a_9    nvidia
     cudf                      22.4.0a0+73.g05d53cf884          pypi_0    pypi
     cupy                      10.1.0           py38h592bde7_0    conda-forge
     cyrus-sasl                2.1.27               h230043b_5    conda-forge
     cython                    0.29.26          py38h709712a_0    conda-forge
     cytoolz                   0.11.2           py38h497a2fe_1    conda-forge
     dask                      2022.1.0+29.g7edfe245          pypi_0    pypi
     dask-cudf                 22.4.0a0+113.g0581975e01          pypi_0    pypi
     dataclasses               0.8                pyhc8e2a94_3    conda-forge
     debugpy                   1.5.1            py38h709712a_0    conda-forge
     decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
     defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge
     distlib                   0.3.4              pyhd8ed1ab_0    conda-forge
     distributed               2022.1.0+21.ge04c5627          pypi_0    pypi
     dlpack                    0.5                  h9c3ff4c_0    conda-forge
     docutils                  0.17.1           py38h578d9bd_1    conda-forge
     double-conversion         3.2.0                h9c3ff4c_0    conda-forge
     entrypoints               0.3             pyhd8ed1ab_1003    conda-forge
     execnet                   1.9.0              pyhd8ed1ab_0    conda-forge
     executing                 0.8.2              pyhd8ed1ab_0    conda-forge
     expat                     2.4.3                h9c3ff4c_0    conda-forge
     fastavro                  1.4.9            py38h497a2fe_0    conda-forge
     fastrlock                 0.8              py38h709712a_1    conda-forge
     filelock                  3.4.2              pyhd8ed1ab_1    conda-forge
     findspark                 2.0.0                    pypi_0    pypi
     flake8                    3.8.3                      py_1    conda-forge
     flit-core                 3.6.0              pyhd8ed1ab_0    conda-forge
     freetype                  2.10.4               h0708190_1    conda-forge
     fsspec                    2022.1.0           pyhd8ed1ab_0    conda-forge
     future                    0.18.2           py38h578d9bd_4    conda-forge
     gcc_impl_linux-64         11.2.0              h82a94d6_12    conda-forge
     gettext                   0.19.8.1          h73d1719_1008    conda-forge
     gflags                    2.2.2             he1b5a44_1004    conda-forge
     glog                      0.5.0                h48cff8f_0    conda-forge
     gmp                       6.2.1                h58526e2_0    conda-forge
     grpc-cpp                  1.42.0               ha1441d3_1    conda-forge
     heapdict                  1.0.1                      py_0    conda-forge
     huggingface_hub           0.4.0              pyhd8ed1ab_0    conda-forge
     hypothesis                6.36.0             pyhd8ed1ab_0    conda-forge
     identify                  2.4.5              pyhd8ed1ab_0    conda-forge
     idna                      3.3                pyhd8ed1ab_0    conda-forge
     imagesize                 1.3.0              pyhd8ed1ab_0    conda-forge
     importlib-metadata        4.10.1           py38h578d9bd_0    conda-forge
     importlib_metadata        4.10.1               hd8ed1ab_0    conda-forge
     importlib_resources       5.4.0              pyhd8ed1ab_0    conda-forge
     iniconfig                 1.1.1              pyh9f0ad1d_0    conda-forge
     ipykernel                 6.7.0            py38he5a9106_0    conda-forge
     ipython                   8.0.1            py38h578d9bd_0    conda-forge
     ipython_genutils          0.2.0                      py_1    conda-forge
     isort                     5.6.4                      py_0    conda-forge
     jbig                      2.1               h7f98852_2003    conda-forge
     jedi                      0.18.1           py38h578d9bd_0    conda-forge
     jinja2                    3.0.3              pyhd8ed1ab_0    conda-forge
     joblib                    1.1.0              pyhd8ed1ab_0    conda-forge
     jpeg                      9e                   h7f98852_0    conda-forge
     jsonschema                4.4.0              pyhd8ed1ab_0    conda-forge
     jupyter_client            7.1.2              pyhd8ed1ab_0    conda-forge
     jupyter_core              4.9.1            py38h578d9bd_1    conda-forge
     jupyterlab_pygments       0.1.2              pyh9f0ad1d_0    conda-forge
     kernel-headers_linux-64   2.6.32              he073ed8_15    conda-forge
     krb5                      1.19.2               hcc1bbae_3    conda-forge
     lcms2                     2.12                 hddcbb42_0    conda-forge
     ld_impl_linux-64          2.36.1               hea4e1c9_2    conda-forge
     lerc                      3.0                  h9c3ff4c_0    conda-forge
     libblas                   3.9.0            13_linux64_mkl    conda-forge
     libbrotlicommon           1.0.9                h7f98852_6    conda-forge
     libbrotlidec              1.0.9                h7f98852_6    conda-forge
     libbrotlienc              1.0.9                h7f98852_6    conda-forge
     libcblas                  3.9.0            13_linux64_mkl    conda-forge
     libclang-cpp11.1          11.1.0          default_ha53f305_1    conda-forge
     libcurl                   7.81.0               h2574ce0_0    conda-forge
     libdeflate                1.8                  h7f98852_0    conda-forge
     libedit                   3.1.20191231         he28a2e2_2    conda-forge
     libev                     4.33                 h516909a_1    conda-forge
     libevent                  2.1.10               h9b69904_4    conda-forge
     libffi                    3.4.2                h7f98852_5    conda-forge
     libgcc-devel_linux-64     11.2.0              h0952999_12    conda-forge
     libgcc-ng                 11.2.0              h1d223b6_12    conda-forge
     libgcrypt                 1.9.4                h7f98852_0    conda-forge
     libgomp                   11.2.0              h1d223b6_12    conda-forge
     libgpg-error              1.42                 h9c3ff4c_0    conda-forge
     libgsasl                  1.10.0               h5b4c23d_0    conda-forge
     liblapack                 3.9.0            13_linux64_mkl    conda-forge
     libllvm11                 11.1.0               hf817b99_2    conda-forge
     libnghttp2                1.46.0               h812cca2_0    conda-forge
     libnsl                    2.0.0                h7f98852_0    conda-forge
     libntlm                   1.4               h7f98852_1002    conda-forge
     libpng                    1.6.37               h21135ba_2    conda-forge
     libprotobuf               3.19.3               h780b84a_0    conda-forge
     librdkafka                1.7.0                hc49e61c_1    conda-forge
     librmm                    22.04.00a220120 cuda11_gdf67616_6    rapidsai-nightly
     libsanitizer              11.2.0              he4da1e4_12    conda-forge
     libsodium                 1.0.18               h36c2ea0_1    conda-forge
     libssh2                   1.10.0               ha56f1ee_2    conda-forge
     libstdcxx-ng              11.2.0              he4da1e4_12    conda-forge
     libthrift                 0.15.0               he6d91bd_1    conda-forge
     libtiff                   4.3.0                h6f004c6_2    conda-forge
     libutf8proc               2.7.0                h7f98852_0    conda-forge
     libuv                     1.43.0               h7f98852_0    conda-forge
     libwebp-base              1.2.2                h7f98852_1    conda-forge
     libzlib                   1.2.11            h36c2ea0_1013    conda-forge
     llvm-openmp               12.0.1               h4bd325d_1    conda-forge
     llvmlite                  0.38.0           py38h4630a5e_0    conda-forge
     locket                    0.2.0                      py_2    conda-forge
     lz4-c                     1.9.3                h9c3ff4c_1    conda-forge
     markdown                  3.3.6              pyhd8ed1ab_0    conda-forge
     markupsafe                2.0.1            py38h497a2fe_1    conda-forge
     matplotlib-inline         0.1.3              pyhd8ed1ab_0    conda-forge
     mccabe                    0.6.1                      py_1    conda-forge
     mimesis                   4.0.0              pyh9f0ad1d_0    conda-forge
     mistune                   0.8.4           py38h497a2fe_1005    conda-forge
     mkl                       2022.0.1           h8d4b97c_803    conda-forge
     msgpack-python            1.0.3            py38h1fd1430_0    conda-forge
     mypy                      0.782                      py_0    conda-forge
     mypy_extensions           0.4.3            py38h578d9bd_4    conda-forge
     nbclient                  0.5.10             pyhd8ed1ab_1    conda-forge
     nbconvert                 6.4.0            py38h578d9bd_0    conda-forge
     nbformat                  5.1.3              pyhd8ed1ab_0    conda-forge
     nbsphinx                  0.8.8              pyhd8ed1ab_0    conda-forge
     ncurses                   6.3                  h9c3ff4c_0    conda-forge
     nest-asyncio              1.5.4              pyhd8ed1ab_0    conda-forge
     ninja                     1.10.2               h4bd325d_1    conda-forge
     nodeenv                   1.6.0              pyhd8ed1ab_0    conda-forge
     notebook                  6.4.8              pyha770c72_0    conda-forge
     numba                     0.55.0           py38h4bf6c61_0    conda-forge
     numpy                     1.21.5           py38h87f13fb_0    conda-forge
     numpydoc                  1.2                pyhd8ed1ab_0    conda-forge
     nvtx                      0.2.3            py38h497a2fe_1    conda-forge
     olefile                   0.46               pyh9f0ad1d_1    conda-forge
     openjpeg                  2.4.0                hb52868f_1    conda-forge
     openssl                   1.1.1l               h7f98852_0    conda-forge
     orc                       1.7.2                h1be678f_0    conda-forge
     packaging                 21.3               pyhd8ed1ab_0    conda-forge
     pandas                    1.3.5            py38h43a58ef_0    conda-forge
     pandoc                    1.19.2                        0    conda-forge
     pandocfilters             1.5.0              pyhd8ed1ab_0    conda-forge
     parquet-cpp               1.5.1                         2    conda-forge
     parso                     0.8.3              pyhd8ed1ab_0    conda-forge
     partd                     1.2.0              pyhd8ed1ab_0    conda-forge
     pathspec                  0.9.0              pyhd8ed1ab_0    conda-forge
     pexpect                   4.8.0              pyh9f0ad1d_2    conda-forge
     pickleshare               0.7.5                   py_1003    conda-forge
     pillow                    8.4.0            py38h8e6f84c_0    conda-forge
     pip                       21.3.1             pyhd8ed1ab_0    conda-forge
     pluggy                    1.0.0            py38h578d9bd_2    conda-forge
     pre-commit                2.17.0           py38h578d9bd_0    conda-forge
     prometheus_client         0.13.0             pyhd8ed1ab_0    conda-forge
     prompt-toolkit            3.0.24             pyha770c72_0    conda-forge
     protobuf                  3.19.3           py38h709712a_0    conda-forge
     psutil                    5.9.0            py38h497a2fe_0    conda-forge
     ptxcompiler               0.2.0            py38hb739d79_0    rapidsai
     ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
     pure_eval                 0.2.2              pyhd8ed1ab_0    conda-forge
     py                        1.11.0             pyh6c4a22f_0    conda-forge
     py-cpuinfo                8.0.0              pyhd8ed1ab_0    conda-forge
     pyarrow                   5.0.0           py38ha746e9d_22_cuda    conda-forge
     pycodestyle               2.6.0              pyh9f0ad1d_0    conda-forge
     pycparser                 2.21               pyhd8ed1ab_0    conda-forge
     pydata-sphinx-theme       0.8.0              pyhd8ed1ab_0    conda-forge
     pydocstyle                6.1.1              pyhd8ed1ab_0    conda-forge
     pyflakes                  2.2.0              pyh9f0ad1d_0    conda-forge
     pygments                  2.11.2             pyhd8ed1ab_0    conda-forge
     pyopenssl                 21.0.0             pyhd8ed1ab_0    conda-forge
     pyorc                     0.5.0                    pypi_0    pypi
     pyparsing                 3.0.7              pyhd8ed1ab_0    conda-forge
     pyrsistent                0.18.1           py38h497a2fe_0    conda-forge
     pysocks                   1.7.1            py38h578d9bd_4    conda-forge
     pytest                    6.2.5            py38h578d9bd_2    conda-forge
     pytest-benchmark          3.4.1              pyhd8ed1ab_0    conda-forge
     pytest-forked             1.4.0              pyhd8ed1ab_0    conda-forge
     pytest-xdist              2.5.0              pyhd8ed1ab_0    conda-forge
     python                    3.8.12          hb7a2778_2_cpython    conda-forge
     python-confluent-kafka    1.7.0            py38h497a2fe_2    conda-forge
     python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
     python-snappy             0.6.0            py38h49bdff1_1    conda-forge
     python_abi                3.8                      2_cp38    conda-forge
     pytorch                   1.10.1          cpu_py38hb2150b6_0    conda-forge
     pytz                      2021.3             pyhd8ed1ab_0    conda-forge
     pyyaml                    6.0              py38h497a2fe_3    conda-forge
     pyzmq                     22.3.0           py38h2035c66_1    conda-forge
     rapidjson                 1.1.0             he1b5a44_1002    conda-forge
     re2                       2021.11.01           h9c3ff4c_0    conda-forge
     readline                  8.1                  h46c0cb4_0    conda-forge
     recommonmark              0.7.1              pyhd8ed1ab_0    conda-forge
     regex                     2022.1.18        py38h497a2fe_0    conda-forge
     requests                  2.27.1             pyhd8ed1ab_0    conda-forge
     rhash                     1.4.1                h7f98852_0    conda-forge
     rmm                       22.04.00a220120 cuda11_py38_ge61e85b_8_has_cma    rapidsai-nightly
     s2n                       1.0.10               h9b69904_0    conda-forge
     sacremoses                0.0.46             pyhd8ed1ab_0    conda-forge
     send2trash                1.8.0              pyhd8ed1ab_0    conda-forge
     setuptools                59.8.0           py38h578d9bd_0    conda-forge
     six                       1.16.0             pyh6c4a22f_0    conda-forge
     sleef                     3.5.1                h9b69904_2    conda-forge
     snappy                    1.1.8                he1b5a44_3    conda-forge
     snowballstemmer           2.2.0              pyhd8ed1ab_0    conda-forge
     sortedcontainers          2.4.0              pyhd8ed1ab_0    conda-forge
     soupsieve                 2.3.1              pyhd8ed1ab_0    conda-forge
     spdlog                    1.8.5                h4bd325d_0    conda-forge
     sphinx                    4.4.0              pyh6c4a22f_1    conda-forge
     sphinx-copybutton         0.4.0              pyhd8ed1ab_0    conda-forge
     sphinx-markdown-tables    0.0.15             pyhd3deb0d_0    conda-forge
     sphinxcontrib-applehelp   1.0.2                      py_0    conda-forge
     sphinxcontrib-devhelp     1.0.2                      py_0    conda-forge
     sphinxcontrib-htmlhelp    2.0.0              pyhd8ed1ab_0    conda-forge
     sphinxcontrib-jsmath      1.0.1                      py_0    conda-forge
     sphinxcontrib-qthelp      1.0.3                      py_0    conda-forge
     sphinxcontrib-serializinghtml 1.1.5              pyhd8ed1ab_1    conda-forge
     sphinxcontrib-websupport  1.2.4              pyhd8ed1ab_1    conda-forge
     sqlite                    3.37.0               h9cd32fc_0    conda-forge
     sre-yield                 1.2                      pypi_0    pypi
     stack_data                0.1.4              pyhd8ed1ab_0    conda-forge
     streamz                   0.6.3              pyh6c4a22f_0    conda-forge
     sysroot_linux-64          2.12                he073ed8_15    conda-forge
     tbb                       2021.5.0             h4bd325d_0    conda-forge
     tblib                     1.7.0              pyhd8ed1ab_0    conda-forge
     terminado                 0.12.1           py38h578d9bd_1    conda-forge
     testpath                  0.5.0              pyhd8ed1ab_0    conda-forge
     tk                        8.6.11               h27826a3_1    conda-forge
     tokenizers                0.10.3           py38hb63a372_1    conda-forge
     toml                      0.10.2             pyhd8ed1ab_0    conda-forge
     toolz                     0.11.2             pyhd8ed1ab_0    conda-forge
     tornado                   6.1              py38h497a2fe_2    conda-forge
     tqdm                      4.62.3             pyhd8ed1ab_0    conda-forge
     traitlets                 5.1.1              pyhd8ed1ab_0    conda-forge
     transformers              4.10.3             pyhd8ed1ab_0    conda-forge
     typed-ast                 1.4.3            py38h497a2fe_1    conda-forge
     typing-extensions         4.0.1                hd8ed1ab_0    conda-forge
     typing_extensions         4.0.1              pyha770c72_0    conda-forge
     ukkonen                   1.0.1            py38h1fd1430_1    conda-forge
     urllib3                   1.26.8             pyhd8ed1ab_1    conda-forge
     virtualenv                20.4.7           py38h578d9bd_1    conda-forge
     wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
     webencodings              0.5.1                      py_1    conda-forge
     wheel                     0.37.1             pyhd8ed1ab_0    conda-forge
     xz                        5.2.5                h516909a_1    conda-forge
     yaml                      0.2.5                h7f98852_2    conda-forge
     zeromq                    4.3.4                h9c3ff4c_1    conda-forge
     zict                      2.0.0                      py_0    conda-forge
     zipp                      3.7.0              pyhd8ed1ab_0    conda-forge
     zlib                      1.2.11            h36c2ea0_1013    conda-forge
     zstd                      1.5.2                ha95c52a_0    conda-forge
     
</pre></details>
```

**Additional context**
N/A

--- a/cpp/src/strings/regex/regcomp.cpp
+++ b/cpp/src/strings/regex/regcomp.cpp
@@ -257,11 +257,13 @@ class regex_parser {
     if (quoted) {
       // treating all quoted numbers as Octal, since we are not supporting backreferences
       if (yy >= '0' && yy <= '7') {
-        yy         = yy - '0';
-        char32_t c = *exprp;
-        while (c >= '0' && c <= '7') {
+        yy          = yy - '0';
+        auto c      = *exprp;
+        auto digits = 1;
+        while (c >= '0' && c <= '7' && digits < 3) {
           yy = (yy << 3) | (c - '0');
           c  = *(++exprp);
+          ++digits;
         }
         return CHAR;
       } else {

--- a/cpp/tests/strings/contains_tests.cpp
+++ b/cpp/tests/strings/contains_tests.cpp
@@ -239,14 +239,14 @@ TEST_F(StringsContainsTests, MatchesIPV4Test)
 
 TEST_F(StringsContainsTests, OctalTest)
 {
-  cudf::test::strings_column_wrapper strings({"AZ", "B", "CDAZEY", ""});
+  cudf::test::strings_column_wrapper strings({"A3", "B", "CDA3EY", ""});
   auto strings_view = cudf::strings_column_view(strings);
   cudf::test::fixed_width_column_wrapper<bool> expected({1, 0, 1, 0});
   auto results = cudf::strings::contains_re(strings_view, "\\101");
   CUDF_TEST_EXPECT_COLUMNS_EQUAL(*results, expected);
-  results = cudf::strings::contains_re(strings_view, "\\101Z");
+  results = cudf::strings::contains_re(strings_view, "\\1013");
   CUDF_TEST_EXPECT_COLUMNS_EQUAL(*results, expected);
-  results = cudf::strings::contains_re(strings_view, "D*\\101\\132");
+  results = cudf::strings::contains_re(strings_view, "D*\\101\\063");
   CUDF_TEST_EXPECT_COLUMNS_EQUAL(*results, expected);
 }
 

